src/navigation/context.py

```diff
@@ -97,6 +106,19 @@ def current_gate(self) -> Optional[Gate]:
             return Gate(post1[0:2], post2[0:2])
         else:
             return None
+    
+    def add_failure_zone(self, failure_zone: FailureZone) -> None:
```

Comment:
do you think this function would make more sense inside of Driver?

Reply:
I hesitated to put the list of failure zones and this function in Driver because it feels like they are more an attribute of the Environment than anything else, and the Driver is just using them to make decisions about driving. That being said, it would be somewhat easier to update the map if this directly went to Driver (rather than calling a function to Driver from Environment every time a change is made or manually checking for changes). Let me know what you think

Reply:
Also, I figured it would be fine to initialize the list of failure zones to empty and have them added one-by-one (and never removed). Do you think this is fine or should there also be support for initializing failure zones with a list (maybe we have a list from prior runs of the rover? idk) and/or removing failure zones?

Reply:
I think that putting it in the Driver is fine since the driver is failure zones more generally are just points that the driver wants to avoid and not really points that anybody else would care about.

Reply:
I think having it start as empty is ok